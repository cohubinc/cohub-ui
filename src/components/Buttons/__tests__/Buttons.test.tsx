import React from "react";
import { mount } from "enzyme";
import { Buttons } from "dist";

describe("Buttons", () => {
  it("renders", () => {
    const wrapper = mount(<Buttons>push me</Buttons>);
    expect(wrapper).toMatchSnapshot();
  });

  describe("Buttons can render other button component via static methods on the class", () => {
    it("renders Primary", () => {
      const wrapper = mount(<Buttons.Primary>push me</Buttons.Primary>);
      expect(wrapper.exists()).toBe(true);
    });

    it("renders Info", () => {
      const wrapper = mount(<Buttons.Info>push me</Buttons.Info>);
      expect(wrapper.exists()).toBe(true);
    });

    it("renders Cancel", () => {
      const wrapper = mount(<Buttons.Cancel>push me</Buttons.Cancel>);
      expect(wrapper.exists()).toBe(true);
    });

    it("renders Outline", () => {
      const wrapper = mount(<Buttons.Outline>push me</Buttons.Outline>);
      expect(wrapper.exists()).toBe(true);
    });

    it("renders Ghost", () => {
      const wrapper = mount(<Buttons.Ghost>push me</Buttons.Ghost>);
      expect(wrapper.exists()).toBe(true);
    });

    it("renders PrimaryGhost ", () => {
      const wrapper = mount(
        <Buttons.PrimaryGhost>push me</Buttons.PrimaryGhost>
      );
      expect(wrapper.exists()).toBe(true);
    });

    it("renders CancelGhost", () => {
      const wrapper = mount(<Buttons.CancelGhost>push me</Buttons.CancelGhost>);
      expect(wrapper.exists()).toBe(true);
    });

    it("renders Blank", () => {
      const wrapper = mount(<Buttons.Blank>push me</Buttons.Blank>);
      expect(wrapper.exists()).toBe(true);
    });
  });
});
