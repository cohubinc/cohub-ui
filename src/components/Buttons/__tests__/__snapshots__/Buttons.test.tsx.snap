// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Buttons renders 1`] = `
<Buttons>
  <Secondary>
    <Base
      backgroundColor="hsl(0, 0%, 100%)"
      style={
        Object {
          "color": "hsl(212, 14%, 35%)",
        }
      }
    >
      <button
        className="CohubButton "
        style={
          Object {
            "backgroundColor": "hsl(0, 0%, 100%)",
            "boxShadow": "0 1px 3px hsla(0, 0%, 0%, 0.1), inset 0px 1px 0px #F2F2F2",
            "color": "hsl(212, 14%, 35%)",
          }
        }
      >
        <div
          className="button-text relative flex items-center"
        >
          <Component
            color="hsl(212, 14%, 35%)"
            style={
              Object {
                "opacity": 1,
                "transition": "opacity 150ms ease-in",
              }
            }
            uppercase={true}
          >
            <span
              className=""
              data-qa="text"
              style={
                Object {
                  "color": "hsl(212, 14%, 35%)",
                  "fontFamily": "Inter",
                  "fontSize": 12,
                  "fontWeight": 400,
                  "opacity": 1,
                  "textAlign": "initial",
                  "textTransform": "uppercase",
                  "transition": "opacity 150ms ease-in",
                }
              }
            >
              <div
                className="flex items-center"
                style={
                  Object {
                    "flexDirection": "row",
                  }
                }
              >
                <span
                  style={
                    Object {
                      "marginLeft": "",
                      "marginRight": "",
                    }
                  }
                >
                  push me
                </span>
              </div>
            </span>
          </Component>
        </div>
      </button>
    </Base>
  </Secondary>
</Buttons>
`;
